src/components/cleaner.ts

```diff
@@ -2,11 +2,35 @@ import * as vscode from 'vscode'
 import * as path from 'path'
 import * as fs from 'fs'
 import glob from 'glob'
-import minimatch from 'minimatch'
 import * as cs from 'cross-spawn'
 
 import type {Extension} from '../main'
 
+
+/**
+ * Removes the duplicate elements. Note that the order of the sequence will not be preserved.
+ */
+function unique(sequence: string[]): string[] {
+    return Array.from(new Set(sequence))
+}
+
+/**
+ * Globs all given patterns into absolute paths. The result will be sorted in
+ * reverse order and all tailing slashes will be stripped.
+ */
+function globAll(globs: string[], cwd: string): string[] {
+    return unique(
+        globs.map(g => glob.sync(g, { cwd }))
+             // Reduce the array of arrays to a single array containing all the elements
+             .reduce((all, curr) => all.concat(curr), [])
+             // Resolve absolute path (tailing slashes will be stripped)
```

Comment:
It seems to me that we do not care about trailing slashes being stripped anymore. That was important prior to this refactoring.

Reply:
> That was important prior to this refactoring.

The old implementation globs all patterns at once. Here we split the globs into two groups and get the results separately. We don't need the tailing slashes anymore.